### ISSUE TITLE
Error `heifload: ignoring nclx profile` when converting to AVIF and resizing

### ISSUE THREAD

**Issue by swissspidy** [kleisauke/wasm-vips#62]
For some images such as https://github.com/swissspidy/media-experiments/blob/f6a7f9f26694b79a25ae2b844d2cc9a344d356a7/tests/e2e/assets/wordpress-logo-512x512.png I am getting this error when trying to convert to AVIF:

<img width="636" alt="Screenshot 2024-01-18 at 22 53 30" src="https://github.com/swissspidy/media-experiments/assets/841956/d1d97e3d-daf2-4131-bcb2-54565098a891">

Existing issues with same error:

* https://github.com/libvips/libvips/issues/1808
* https://github.com/libvips/libvips/issues/1888

Weird thing is that this is a PNG / AVIF situation, so not sure why HEIF would be involved here.

Here's a playground repro:

https://wasm-vips.kleisauke.nl/playground/?modules=jxl-heif&deflate=rVZtb9s2EP4rN3-RUjiSk2HA4MUJkGIBAgwoNnTYhzhAaekkc5BIjaTiuq3_-44UKdmak3hFAsSieC_P3XPHEx8mmRTaQCPK27YoUMEC2IZxA_HoWaDJ1jFEa2MaPU9TxTZJyc26XbUaFXkxKEySyTrVG661bni-TWvMOTvHzw0qXpNcpzXjIjWoaYmXmDKtkZYbqfJGodbnlSzl-U8Xl5_pP6GoIjiDs2RVyVVsF0wptu0ijc9-WYouevbEi1H4JHhCZe5rVuKdVDUzcZ_jYEiQbWV6I3rlX9DZxIPPKXyFDc_Neg4_zmZTWCMv18a_ZEo2cyhYpRF2UzCqRes-ANjAyb3ADdzaHB46xEfr02wbnEPELVxq4SLYDaFZyz__-I2M6TfJFDKDH1Z_Y2boPbbSQZfXJenlMmstyV751wrtW0wAZWR16ZlolZGm9017vU2Jxhvcbu_zOJKtaVoTEeNNgyJ_v-ZVHpOHLjmmtyKDohWZ4VIcI5u0zKqjzy7_aVnFzZawZ8nPl0txBl_tdgif7Ej0xBudOCcJEXanZO0rDZ0n6MCDmWwsuCZD52tpfp9DwHkHF1Qdu72zNsGiNX2bONBko7jBjzLgRImrw7T37RC9dd8svZukiytEVSETbXOTuM60OwpNq4Q3pL3dEe72e25EWieiJdCfrpky76nbKADXbiMS_8OGtaX2yuVGRAMT9Kzw_1LuIfwxCCcAdoFG7zhN4b7wCSVeh2v4gkomQz5BshhpfvtGs8av0w6Jzv67oOU29jLotv9yar2rXsl4Op4B8354QZA_BJE9yoTo6TvCEA27eiUYrw45mu7HMm6dHaAdDQ7JDgdYLPpYDgFDxG53sVfvG4iYsdOVxBFQSTNaK4x8Em8dqg-HinnHVdf2tgvBrFHQjLFPN_UaLkpXVqf7cU2VZlUlN5rcomKGxED9bdU1UgPlsOKmZg2VlkrmvEilaJ5Bbr8N2iXfp2T56nrgKjRFz1ToJF_8sNk3WGd3Par5oLY0NwcNsy-Z97jpgNs34dA9TnncZN8VyQtN7xI8jO5YIIeFewt2XufgaFTD8Xszdl4AfY24wM7bHhPnyg6gCgs7V2YexnWsO5qoovEpf4AZPO43sDeOw2fIxX9-wCOpp3B5WOIORNmsT8Z4HmFMkM2KqnZ6UheHgJ1tyMkXp4fsO-mZrFaSHNSnoryE8VzhOxvrNHbVo5kmm1DrjpL-LUTbTcaDa8eJl4jRbeW12wPN0A-i2kJGQxRYQYTDJ-f-k_2GCgmVFCVtrtAOVrpz5274nnDpmEwnVzl_Ap4vlpPuXrecXF-ltHe9FCSln8d_AQ

**Comment by kleisauke** [comment]
Hi @swissspidy,

It's safe to ignore this warning in this case. Upstream PR https://github.com/libvips/libvips/pull/3834 would remove this redundant warning.

> Weird thing is that this is a PNG / AVIF situation, so not sure why HEIF would be involved here.

HEIF is a container, which supports HEVC and AV1 codecs used to create HEIC and AVIF images respectively. libvips leverages libheif to load/save these images.

**Comment by swissspidy** [comment]
Understood, thanks!

Is there a way to set the desired log level to hide info/warning/error messages?

**Comment by kleisauke** [comment]
The `VIPS_WARNING` environment variable is supposed to suppress these warning messages, for example:
```js
const vips = await Vips({
  preRun: (module) => {
    // Hide warning messages
    module.ENV.VIPS_WARNING = 1;
  }
});

const im = vips.Image.newFromFile('x.avif');
im.writeToFile('x.png');

// Memory management
im.delete();
```

But, for some reason, it doesn't work for warnings emitted from side modules (i.e. `vips-heif.wasm`), which is curious. 

I spent a couple of hours debugging to no avail, when I build wasm-vips without loadable modules, i.e.:
```console
$ npm run build -- --disable-modules
```

The above program just works. :man_shrugging:

For now, I can recommend this workaround:
```js
// Empty log handler
const log = (type, text) => void 0; // Or: console.log(`${type}: ${text}`);
const vips = await Vips({
  print: (text) => log('stdout', text),
  printErr: (text) => log('stderr', text),
});
```

**Comment by kleisauke** [comment]
v0.0.8 is now available and provides libvips v8.15.2.
